worker/backend/rootfs_manager.go

```diff
@@ -0,0 +1,70 @@
+package backend
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . RootfsManager
+
+// RootfsManager is responsible for mutating the rootfs of a container.
+//
+type RootfsManager interface {
+	// SetupCwd mutates the root filesystem to guarantee the presence of a
+	// directory to be used as `cwd`.
+	//
+	SetupCwd(containerSpec *specs.Spec, cwd string) (err error)
+}
+
+// RootfsManagerOpt defines a functional option that when applied, modifies the
+// configuration of a rootfsManager.
+//
+type RootfsManagerOpt func(m *rootfsManager)
+
+// WithMkdirAll configures the function to be used for creating directories
+// recursively.
+//
+func WithMkdirAll(f func(path string, mode os.FileMode) error) RootfsManagerOpt {
+	return func(m *rootfsManager) {
+		m.mkdirall = f
```

Comment:
curious, when would you use a different mkdir function?

Reply:
gooood question - the rationale was that I wanted to be able to verify that `mkdirall` was  called only a certain number of times:

https://github.com/concourse/concourse/blob/54991b2f1f17bd7d229a3ebbe6d6c16f3ff495a3/worker/backend/rootfs_manager_test.go#L41-L48

it could also be used to inject errors (I actually should've done that in the tests to test some failure scenarios 🤔)